src/app/components/ExpandableSearch/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { FC, PropsWithChildren } from 'react'
+import { styled, useTheme } from '@mui/material/styles'
+import Box from '@mui/material/Box'
+import { Search } from '../Search'
+import useMediaQuery from '@mui/material/useMediaQuery'
+
+const ExpandableSearchBox = styled(Box)(() => ({
+  position: 'relative',
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  width: '100%',
+}))
+
+const ContentBox = styled(Box)(({theme}) => ({
+  flex: '0 1 calc(100% - 50px)',
+  [theme.breakpoints.up('sm')]: {
+    flex: '0 1 100%',
+  },
+}))
+
+const SearchBox = styled(Box)(() => ({
+  flex: '0 1 50px',
```

Comment:
All this flex magic should not be needed. You already have `justify-content: space-between`

Reply:
Found no other smart way to make it work. As the Search is positioned absolutely in expanded version, rather than hardcoding values.

Reply:
```tsx
const SearchWrapper = styled(Box)(() => ({
  height: '47px',
  width: '50px', // Reserve space for absolutely positioned search icon
}))
```